packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1426,6 +1463,85 @@ export class TransactionController extends BaseController<
 
     return dappSuggestedGasFees;
   }
+
+  /**
+   * Validates and adds external provided transaction to state.
+   *
+   * @param transactionMeta - Nominated external transaction to be added to state.
+   */
+  private async addExternalTransaction(transactionMeta: TransactionMeta) {
+    const { networkId, chainId } = this.getChainAndNetworkId();
+    const { transactions } = this.state;
+    const fromAddress = transactionMeta?.transaction?.from;
+    const sameFromAndNetworkTransactions = transactions.filter(
+      (transaction) =>
+        transaction.transaction.from === fromAddress &&
+        transactionMatchesNetwork(transaction, chainId, networkId),
+    );
+    const confirmedTxs = sameFromAndNetworkTransactions.filter(
+      (transaction) => transaction.status === TransactionStatus.confirmed,
+    );
+    const pendingTxs = this.state.transactions.filter(
```

Comment:
Should this be referencing `sameFromAndNetworkTransactions` also?

Reply:
Sorry, missed that, fixed in https://github.com/MetaMask/core/pull/1625/commits/293501b6a34fcde1be99560f1c2f65766cb28a1d